packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -159,7 +159,7 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
                 table: version_column.table,
                 access: ColumnAccess::Table { column: "base_uri" },
             },
-            TableName::DataTypes => unreachable!(),
+            _ => unreachable!(),
```

Comment:
Too far from the diff, but if this is unreachable, why do we have the comment above. It sounds like it doesn't depend on the table name, but it's always TypeIds?

Reply:
With support added for `Entity`, a match arm on `TableName::Entities` will be added. The column for `TypeIds` is `"base_uri"` but for entities it's `entity_id`: https://github.com/hashintel/hash/blob/ce6c57480aa8a67ebca788ed299e322e7ef31346/packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs#L155-L168

Reply:
Makes sense, thank you. Do we maybe not want
```
 _ => unreachable!(),
```
then?

Having this will mean we won't get any compile errors for the new table names. But if we explicitly added the variants then when we add new ones, we will get a compile error. If `TableName::Entities` already exists then, then we could have it as `unimplemented!` or something?

Reply:
We spoke offline about this. I agree that having all variants here is more robust, but the `TableName` enum will get quite big with more functionality added. For now we will leave it as it is.